pilot/pkg/bootstrap/server.go

```diff
@@ -443,7 +449,47 @@ func (c *mockController) Run(<-chan struct{}) {}
 
 // initConfigController creates the config controller in the pilotConfig.
 func (s *Server) initConfigController(args *PilotArgs) error {
-	if args.Config.Controller != nil {
+	mcpServerAddrs := args.MCPServerAddrs
+	if len(mcpServerAddrs) > 0 {
+		// TODO: log random uuid and assign to this pilot node
+		clientNodeID := "some-id"
+		supportedTypes := make([]string, len(model.IstioConfigTypes))
+		for _, model := range model.IstioConfigTypes {
+			supportedTypes = append(supportedTypes, model.MessageName)
+		}
+		log.Infof("mcp client initialized with supported types: %+v\n", supportedTypes)
+		logger := log.RegisterScope("coredatamodel", "mcp controller debugging", 0)
+		mcpController := coredatamodel.NewController(logger)
+		s.addStartFunc(func(stop <-chan struct{}) error {
+			go mcpController.Run(stop)
+			return nil
+		})
+
+		for _, addr := range mcpServerAddrs {
+			conn, err := grpc.Dial(addr, grpc.WithInsecure())
```

Comment:
Add a TODO here - obviously we won't ship with insecure.

Reply:
👍